src/schema/document.rs

```diff
@@ -191,6 +191,34 @@ impl Document {
     pub fn get_first(&self, field: Field) -> Option<&Value> {
         self.get_all(field).next()
     }
+
+    /// Serializes stored field values.
+    pub fn serialize_stored<W: Write>(&self, schema: &Schema, writer: &mut W) -> io::Result<()> {
```

Comment:
can we make this a function in segment_writer?

Reply:
I think the abstraction is better like this, so the store doesn't need to know about `Document` internals like FieldValues

Reply:
Ah yes you are right!